integration/helpers/client_provider.py

```diff
@@ -115,6 +116,16 @@ def cloudwatch_event_client(self):
                 self._cloudwatch_event_client = boto3.client("events")
         return self._cloudwatch_event_client
 
+    @property
+    def cloudwatch_client(self):
+        """
+        CloudWatch Client
+        """
+        with self._lock:
```

Comment:
Why do we have a lock? First reaction is this seems odd but don't understand why we would need to do this.

Reply:
This client provider may call by multi threads, so we add lock here to prevent concurrency issues.

Reply:
I wonder if there is a different way but not critical for this. Thanks for explaining.